pandas/tests/frame/methods/test_update.py

```diff
@@ -140,6 +140,22 @@ def test_update_datetime_tz(self):
         expected = DataFrame([pd.Timestamp("2019", tz="UTC")])
         tm.assert_frame_equal(result, expected)
 
+    def test_update_datetime_tz_in_place(self, using_copy_on_write, warn_copy_on_write):
+        # https://github.com/pandas-dev/pandas/issues/56227
+        result = DataFrame([pd.Timestamp("2019", tz="UTC")])
+        orig = result.copy()
+        view = result[:]
+        with tm.assert_produces_warning(
```

Comment:
Can you use ``assert_cow_warning`` here? That's optimised for these types of warnings and requires only ``warn_copy_on_write`` as input